[feature #2528] separate zh_hant to zh_hant_tw, zh_hant_hk and zh_han_sg

foreseaz opened the pull request:
separate zh_hant to zh_hant_tw, zh_hant_hk and zh_hant_sg, update zh_hant

Fixes #2528 

### Summary:

[comment]: # Finish 5 main tasks for Chinese translations:
1. Update `zh_hant` to most updated `en` and `zh_hans`.
2. Separate `zh_hant` to `tw`, `hk`, `sg` based on [OpenCC standard](https://github.com/BYVoid/OpenCC).
3. Fix layout problems of [中英文之間需要增加空格, Space needed between English and Chinese Characters](https://github.com/sparanoid/chinese-copywriting-guidelines#%E4%B8%AD%E8%8B%B1%E6%96%87%E4%B9%8B%E9%96%93%E9%9C%80%E8%A6%81%E5%A2%9E%E5%8A%A0%E7%A9%BA%E6%A0%BC) for `zh_hans`, `zh_hant`, `zh_hant_*`.
4. Make consistently using of `你(You)` instead of `您 (Sir/Madam)`
5. update `bots/browse/translations.js` `bots/console/translations.js` `bots/mailman/translations.js` `bots/transactor/translations.js` for the missing three Traditional Chinese objects

### Steps to test:
- Open directory `/src/status_im/translations/`
- Kindly find `zh_hans.cljs`, `zh_hant.cljs`, `zh_hant_hk.cljs`, `zh_hant_sg.cljs`, `zh_hant_tw.cljs` for details.

status: ready

foreseaz committed:
[feature #2528]
 - separate zh_hant to zh_hant_tw, zh_hant_hk and zh_hant_sg
 - add new locales `zh-hant-sg`, `zh-hant-hk`, `zh-hant-tw`
 - fix zh_hant and zh_hans typo and layout
 - update `bots/browse/translations.js` `bots/console/translations.js`
`bots/mailman/translations.js` `bots/transactor/translations.js`

foreseaz reviewed (commented):
@andytudhope could you please review the PR. Thanks so much.

andytudhope commented:
Will do @foreseaz, thank you. We just merged this PR: https://github.com/status-im/status-react/pull/2510, so you can pull latest `develop`, follow the instructions to add your 3 new locales and test if all the correct keys (i.e. commands) are there. Then I can get @ninadou to look over the actual translations for us. If you have issues, let me know, and I can help run those tests for you.

foreseaz commented:
@andytudhope , I just added the keys [here](https://github.com/status-im/status-react/pull/2552/files#diff-640fd584c0f806e87bf9fbded11fef3b) based on PR #2510 , and also updated `bots/browse/translations.js` `bots/console/translations.js` `bots/mailman/translations.js` `bots/transactor/translations.js` for the missing three Traditional Chinese objects. Could you run the tests for me? Thanks!

andytudhope commented on src/status_im/i18n.cljs:
```diff
@@ -115,9 +118,9 @@
                              :zh-hans-cn  zh-hans/translations
```
Please organise the Chinese section here like so:
```
                             :zh          zh-hans/translations
                             :zh-hans     zh-hans/translations
                             :zh-hans-cn  zh-hans/translations
                             :zh-hans-mo  zh-hans/translations
                             :zh-hant     zh-hant/translations
                             :zh-hant-tw  zh-hant-tw/translations
                             :zh-hant-sg  zh-hant-sg/translations
                             :zh-hant-hk  zh-hant-hk/translations
                             :zh-hant-cn  zh-hant/translations
                             :zh-hant-mo  zh-hant/translations
                             :zh-wuu      zh-wuu/translations
                             :zh-yue      zh-yue/translations
```

ninadou replied:
It said "we went looking everywhere, but couldn’t find those commits". So where  can I find the latest translations?

foreseaz replied:
@ninadou I use `add --amend && push -f` to make the changes into one single commit, so the hash of commit at this message may not be available now, how about go [here](https://github.com/status-im/status-react/pull/2552/files)?

foreseaz replied:
Hey @ninadou , sorry for the disturb, any updates that I should fix here?

andytudhope commented:
If you then scroll down a bit in `i18n.cljs` to here, and add in your new supported languages in a sensisble order like so:
```
(def supported-locales (spec/assert ::locales #{:fr
                                                :zh
                                                :zh-hans
                                                :zh-hans-cn
                                                :zh-hans-mo
                                                :zh-hant
                                                :zh-hant-sg
                                                :zh-hant-hk
                                                :zh-hant-tw
                                                :zh-hant-mo
                                                :zh-hant-cn
                                                :sr-RS_#Cyrl
                                                :en
                                                :de
                                                :lt
                                                :sr-RS_#Latn
                                                :sr}))
```

And run `lein test-cljs` you will see that they all pass! 🎉 And we've added a bunch of new languages to our list! Amazing, amazing work 😄  So, next steps will be for @ninadou to review the content, for me to get a bounty assigned to this, and then we can merge!

foreseaz commented:
@andytudhope Just re-organise the Chinese section locale and add new supported locales, thank you for being so supportive!

ninadou commented on resources/js/bots/console/translations.js:
```diff
@@ -102,6 +102,87 @@ I18n.translations = {
+        password_description: '密碼',
+        password_placeholder: '輸入密碼',
+        password_placeholder2: '請重新輸入密碼以確認',
+        password_error: '密碼應不少於6個字元。',
```
password_error: '密碼應不少於6個字符。',

oskarth commented:
@andytudhope Can this be merged?

andytudhope commented:
@oskarth I am just trying to get a sensible reply out of @ninadou about the actual translated content. As soon as she comes back to me with the OK we can merge - have tested and everything works well.

ninadou commented:
I have pointed out some incorrect or confusing translations. But the author didn't reply me or change the codes. But most of them are correct, so I don't know what to do next...

andytudhope commented:
@ninadou the only review you have done here is the comment `It said "we went looking everywhere, but couldn’t find those commits". So where can I find the latest translations?`

I see no translation change requests - please it's important that you do all reviews here in the open on github so that we can track it and don't get blocked by stuff like this. Where and what, specifically, did you ask to be changed so that @foreseaz can do it and we can merge.

ninadou commented on resources/js/bots/console/translations.js:
```diff
@@ -78,16 +78,16 @@ I18n.translations = {
-        password_placeholder2: '重新鍵入您的密碼',
+        password_placeholder: '鍵入你的密碼',
+        password_placeholder2: '重新鍵入你的密碼',
         password_error: '密碼不得短於6個字元。',
```
password_error: '密碼不得短於6個字符。',

ninadou commented on resources/js/bots/console/translations.js:
```diff
@@ -102,6 +102,87 @@ I18n.translations = {
+        password_description: '密碼',
+        password_placeholder: '輸入密碼',
+        password_placeholder2: '請重新輸入密碼以確認',
+        password_error: '密碼應不少於6個字元。',
```
password_error: '密碼應不少於6個字符。',

ninadou commented on resources/js/bots/console/translations.js:
```diff
@@ -159,7 +240,7 @@ I18n.translations = {
     },
     'zh-wuu': {
         phone_title: '发送电话号码',
```
phone_title: '發送電話號碼',

ninadou commented on resources/js/bots/transactor/translations.js:
```diff
@@ -74,55 +74,94 @@ I18n.translations = {
         request_description: '要求付款',
         request_requesting: '正在请求 ',
 
         validation_title: '金額',
```
validation_title: '金额',

ninadou commented on resources/js/bots/transactor/translations.js:
```diff
@@ -74,55 +74,94 @@ I18n.translations = {
         send_description: '發送付款',
 
-        request_title: '徵求ETH',
+        request_title: '徵求 ETH',
```
request_title: '請求 ETH',

ninadou commented on resources/js/bots/transactor/translations.js:
```diff
@@ -74,55 +74,94 @@ I18n.translations = {
 
-        request_title: '徵求ETH',
+        request_title: '徵求 ETH',
         request_description: '徵求付款',
```
request_description: '請求付款',

ninadou commented on resources/js/bots/transactor/translations.js:
```diff
@@ -74,55 +74,94 @@ I18n.translations = {
-        request_title: '徵求ETH',
+        request_title: '徵求 ETH',
         request_description: '徵求付款',
         request_requesting: '徵求中 ',
```
request_requesting: '請求中',

ninadou commented on src/status_im/translations/zh_hans.cljs:
```diff
@@ -85,14 +85,14 @@
    :update-status                         "更新你的状态..."
    :add-a-status                          "添加状态..."
-   :status-prompt                         "设置你的 Status。使用 #hastags 可以帮助其他人了解您，并且谈论您正在想些什么。"
+   :status-prompt                         "设置你的 Status。使用 #hastags 可以帮助其他人了解你，并且谈论你正在想些什么。"
```
:status-prompt                         "设置你的 Status。使用 #hashtags 可以帮助其他人了解你，并且谈论你正在想些什么。"

ninadou commented:
@andytudhope Can you see my comment now? I left the comments days ago but I didn't submit them. So I thought you can see them. Sorry for stuck here...

andytudhope commented:
@foreseaz been pinging you relentlessly on Riot, but no response? Any chance that you could handle the reviews Nina has suggested so that we can merge this PR?

foreseaz commented:
@andytudhope been offline on Riot these days, I will update asap.

foreseaz commented:
@ninadou Updated according to reviews, thanks for the support.
@andytudhope Sorry for the late response on Riot (so many other IMs distracting 😞), I will try to keep tracking on Riot constantly.

Just updated and please review. Thanks, guys.
